File: org.summer.view.widget/src/org/summer/view/widget/data/BindingMode.java
package org.summer.view.widget.data;
/// <summary> This enum describes how the data flows through a given Binding
    /// </summary> 
    public enum BindingMode
    {
        /// <summary> Data flows from source to target and vice-versa </summary>
        TwoWay, 
        /// <summary> Data flows from source to target, source changes cause data flow </summary>
        OneWay, 
        /// <summary> Data flows from source to target once, source changes are ignored </summary> 
        OneTime,
        /// <summary> Data flows from target to source, target changes cause data flow </summary> 
        OneWayToSource,
        /// <summary> Data flow is obtained from target property default </summary>
        Default
    } 

    /// <summary> This enum describes when updates (target-to-source data flow) 
    /// happen in a given Binding. 
    /// </summary>
    

File: summer/WebContent/js/system/windows/data/BindingMode.js
/**
 * Second Check 12-27
 * BindingMode
 */

define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	
    var BindingMode = declare("BindingMode", Object,{ 
    });	
    
    /// <summary> Data flows from source to target and vice-versa </summary>
    BindingMode.TwoWay = 0;
    /// <summary> Data flows from source to target, source changes cause data flow </summary>
    BindingMode.OneWay = 1;
    /// <summary> Data flows from source to target once, source changes are ignored </summary> 
    BindingMode.OneTime = 2;
    /// <summary> Data flows from target to source, target changes cause data flow </summary> 
    BindingMode.OneWayToSource = 3;
    /// <summary> Data flow is obtained from target property default </summary>
    BindingMode.Default = 4;
    
//    BindingMode.Type = new Type("BindingMode", BindingMode, [Object.Type]);
    return BindingMode;
}); 
